coprocessor/fhevm-engine/fhevm-listener/src/database/tfhe_event_propagate.rs

```diff
@@ -46,22 +46,31 @@ pub struct Database {
     pool: sqlx::Pool<Postgres>,
     tenant_id: TenantId,
     chain_id: ChainId,
+    bucket_cache: std::sync::Arc<
+        tokio::sync::RwLock<lru::LruCache<Handle, PrimitiveDateTime>>,
+    >,
 }
 
 impl Database {
     pub async fn new(
         url: &str,
         coprocessor_api_key: &CoprocessorApiKey,
         chain_id: ChainId,
+        bucket_cache_size: u16,
     ) -> Self {
         let pool = Self::new_pool(url).await;
         let tenant_id =
             Self::find_tenant_id_or_panic(&pool, coprocessor_api_key).await;
+        let bucket_cache =
+            std::sync::Arc::new(tokio::sync::RwLock::new(lru::LruCache::new(
+                std::num::NonZeroU16::new(bucket_cache_size).unwrap().into(),
```

Comment:
Maybe have minimal safe size instead of an error (could be 1).
e.g. `bucket_cache_size.max(MINIMAL_BUCKET_CACHE_SIZE)`
Also bad size like 0 could be checked by command line, so the error is clearer.
How do you disable the cache ? (I think not being able to disable is ok but you know better)